### PR TITLE
Bug 1644982 - If a newer diff exists on the patch, skip Phabricator comment publication

### PR DESCRIPTION
**Pull request by EvaBardou**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1644982

### CODE REVIEW

File: bot/tests/test_reporter_phabricator.py [hunk 1]
```diff
@@ -997,3 +998,36 @@ def test_phabricator_external_tidy(mock_phabricator, phab, mock_try_task, mock_t
...
+    reporter.publish(
+        [],
+        revision,
+        [],
+        [doc_notice],
+    )
+
+    # Check the comment hasn't been posted
+    assert phab.comments[51] == []
+
```
Comment (marco-c) [inline bot/tests/test_reporter_phabricator.py:1031]: Can you make sure the issues are still published as Harbormaster results?

Comment (marco-c) [reply bot/tests/test_reporter_phabricator.py:1031]: (I mean as part of the test)

**Review by abpostelnicu — DISMISSED** [review]
I will approve this, but please do the modifications that @marco-c suggested in the test.

File: bot/tests/test_reporter_phabricator.py (hunk 1 above)
Comment (EvaBardou) [reply bot/tests/test_reporter_phabricator.py:1074]: done!

**Review by marco-c — APPROVED** [review]

**Pull request merged**